code/unit_tests/items.dm

```diff
@@ -36,13 +47,72 @@
 			if(!T.run_test(obj_test_instances[objpath]))
 				failures += "Failed volatile test: [objpath], [T.type]"
 
+	// Run harddel checks on each object.
+	for(var/objpath in obj_test_instances)
+		var/obj/item/curr_obj = obj_test_instances[objpath]
+		if(QDELETED(curr_obj)) // curse you, QDEL_IN
+			continue
+		qdel(curr_obj, force = TRUE)
+		obj_test_instances -= objpath // have to delete all references for it to GC, duh
```

Comment:
Does it clear out faster if you remove the references *prior* to qdeling it?

Reply:
It's going to take at least one queue run no matter what.